pkg/reconciler/pipeline_templates.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+)
+
+func LoadPipelineTemplates() (*dfv1.Templates, error) {
```

Comment:
Use the same way as `LoadConfig` to achieve hot reloading?

Reply:
sure, i was struggling to get viper to unmarshal the yaml the way it does for config but I can still use viper to watch it but not unmarshal.  or if we need viper to unmarshal, I can look at why viper was unmarshaling it to empty Templates.

i'm slightly concerned with thread-safety.  does it make sense to add read/write locking?